service/workflow/models.py

```diff
@@ -1,280 +1,294 @@
-
-                sinks = []
-                stale_tokens = []
-
-                # Fire the transitions
-                for transition in fireable_transitions:
-                    print(f'\n****************************************************************')
-                    print(f'Evaluating at transition {transition}...')
-                    print(f'################################################################\n')
-                    for case in transition.fireable_cases():
-                        print(f'Evaluating case {case}')
-                        tokens = []
-                        for token in transition.all_tokens():
-                            if token.case is not None:
-                                if token.case.id == case.id:
-                                    print(token)
-                                    print(token.__dict__)
-                                    tokens.append(token)
-                        print(tokens)
 
-                        try:
-                            sink = transition.fire([token.__dict__ for token in tokens], case)
-                        except Exception as e:
-                            print("transition did not fire")
-                            raise e
-
-                        print(f'\nNew Tokens: {sink}')
-                        # Mark tokens as stale as transitions that depend on them fire.
-                        # Defer deletion until after this round of transitions have fired.
-                        stale_tokens = list(chain(stale_tokens, tokens))
-                        # Defer creation until after this round of transitions have fired.
-                        #import ipdb; ipdb.set_trace()
-                        sinks = list(chain(sinks, sink))
-
-                # All transitions have fired, delete stale tokens
-                print(f'\nDeleting stale tokens')
-                for token in stale_tokens:
-                    print(f'Token {token}')
-                    token.delete()
-
-                # All transitions have fired, now create tokens
-                print('\nSaving newly created tokens...')
-                for sink in sinks:
-                    print(f'Token {sink}')
-                    Token(**sink).save()
-
-                print('')
 
-    def __str__(self):
-        return self.name
+def print_section_header(string):
+    print('\n\n\033[38;2;230;223;196m\033[1m################################################################################')
+    print('')
+    print(f'{string}')
+    print('################################################################################\033[0m\n\n')
+
+
+def print_subheading_with_context(subheading, context):
+    print('\n\n\033[38;2;230;223;196m\033[1m--------------------------------------------------------------------------------')
+    print(f'\n{subheading}: \033[0m{context}\033[0m\n')
+
+
+class ConditionException(Exception):
+    pass
 
 
+class Workflow(Model):
```

Comment:
The models are pretty much all different. I think message is still pretty similar though